addons/website/models/website.py

```diff
@@ -616,20 +627,32 @@ def set_images(images):
                 # want the xpath error to break the configurator feature
                 logger.warning(e)
 
-        # Load suggestion from iap for selected pages
-        custom_resources = self._website_api_rpc(
-            '/api/website/2/configurator/custom_resources/%s' % kwargs['industry_id'],
-            {'theme': theme_name, }
-        )
-
         # Update pages
+        image_processings_to_do = {}
         requested_pages = list(pages_views.keys()) + ['homepage']
         snippet_lists = website.get_theme_snippet_lists(theme_name)
         for page_code in requested_pages:
-            configure_page(page_code, snippet_lists.get(page_code, []), pages_views, cta_data)
+            configure_page(page_code, snippet_lists.get(page_code, []), pages_views, cta_data, image_processings_to_do, done_image_processings)
+
+        if not done_image_processings:
+            # If image processing needed:
+            # - another call to configurator_apply will happen later => website must not be created yet
+            # - but images must have been obtained before being processed
+            if image_processings_to_do:
+                self.env.cr.rollback()
```

Comment:
Isn't there a way to maybe return early when we detect that changes need to be done on images ? Maybe we can check if images need to be processed before we ever build a page or change a model.

I think we might have talked about that earlier and there might have been a reason, if that's the case, I wouldn't mind being reminded of it 😄